### DISCUSSION TITLE
XML support

### DISCUSSION THREAD

**Discussion by pionl** [CuyZ/Valinor#128] [Ideas]
Hi,

just wondering, how hard would be to add XML support?

**Comment by romm** [comment]
Hi @pionl, I guess you are talking about converting a XML string/file to an object structure, right?

I don't have much experience in XML "conversion", but what I can tell you is that this library will work with any plain PHP array. It means that if you can convert an XML into an array of scalar values, it can then be given to the mapper like any other array.

XML is more powerful than low-level configuration syntax like JSON or YAML, I'm not sure what issues we could encounter during a conversion to an array.

If you need this feature and know how that would be possible, you can take a look at [JsonSource](https://github.com/CuyZ/Valinor/blob/master/src/Mapper/Source/JsonSource.php) for inspiration; and please don't hesitate to open a PR!

**Reply by pionl** [reply]
Hi @romm , thank you for the answer. I will think about it. XML in PHP is very wierd without "schema" (I have experience only with SimpleXMLElement and sometimes it is a pain to work with. Also all "values" are strings + attributes...).

I've found this possible "solution" https://stackoverflow.com/a/20431742, but not sure if it will work in a "good" way.

If end up making it I will let you know :)

Have a great day,
